include/miopen/miopen.h

```diff
@@ -4666,6 +4666,120 @@ miopenReduceTensor(miopenHandle_t handle,
+                                                const miopenTensorDescriptor_t descriptor);
+
+miopenStatus_t miopenSetProblemOperatorDescriptor(miopenProblem_t problem,
+                                                  const void* operatorDesc,
+                                                  miopenProblemDirection_t direction);
+
+/**
+ * The miopenSearchOptions will allow the user to configure how search will be
+ * used. This can include things like exhaustiveSearch, and ordering. In the
+ * future, this can support setting different find mode options.
+ */
+MIOPEN_DECLARE_OBJECT(miopenSearchOptions);
+
+miopenStatus_t miopenCreateSearchOptions(miopenSearchOptions_t* options);
+miopenStatus_t miopenDestroySearchOptions(miopenSearchOptions_t options);
+miopenStatus_t miopenSetSearchOption(miopenSearchOptions_t options,
+                                     miopenSearchOptionName_t optionName,
+                                     size_t valueSize,
+                                     void* value);
+
+MIOPEN_DECLARE_OBJECT(miopenSolution);
+
+/**
+ * Finds solutions to a problem by running different applicable solutions. Memory is automatically
+ * allocated.
+ */
+miopenStatus_t miopenFindSolutions(miopenHandle_t handle,
+                                   miopenProblem_t problem,
+                                   miopenSearchOptions_t options,
+                                   miopenSolution_t* solutions,
+                                   size_t* numSolutions,
+                                   size_t maxSolutions);
+
+/**
+ * Runs the solution using the passed in buffers. Corresponding tensor
+ * descriptors are passed in so if a user would like to dynamically change
+ * the sizes, they can do so. If dynamic changes are not supported(or are
+ * invalid), we should return an error to the user.
+ */
+miopenStatus_t miopenRunSolution(miopenHandle_t handle,
+                                 miopenSolution_t solution,
+                                 size_t nInputs,
+                                 miopenTensorName_t* names,
+                                 miopenTensorDescriptor_t* descriptors,
+                                 void** buffers,
+                                 void* workspace,
+                                 size_t workspaceSize);
+
+miopenStatus_t miopenDestroySolution(miopenSolution_t solution);
+
+miopenStatus_t miopenLoadSolution(miopenSolution_t* solution, const char* data, size_t size);
+
+miopenStatus_t miopenSaveSolution(miopenSolution_t solution, char* data);
+
+miopenStatus_t miopenGetSolutionSize(miopenSolution_t solution, size_t* size);
+
+miopenStatus_t miopenGetSolutionAttribute(miopenSolution_t solution,
+                                          miopenSolutionAttribute_t solutionAttribute,
+                                          size_t valueSize,
+                                          void* value,
+                                          size_t* valueSizeRet);
```

Comment:
Same here. There should be a function for each attribute.

Reply:
Yea we should try to avoid `void*` parameters in our code wherever possible. I dont see an issue with having a function instead of an enum for each attribute.